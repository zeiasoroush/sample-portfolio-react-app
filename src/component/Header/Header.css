nav {
  top: 15%;
  text-align: center;
  background: rgba(0, 0, 0, 0.75);
  position: fixed;
  z-index: 100;
  height: 90%;
  left: 0;
  width: 15%;
  font-weight: 300;
  font-size: 1rem;
}
nav em {
  font-style: normal;
  font-size: 13px;
  text-transform: uppercase;
  display: block;
  margin-top: 15px;
  color: #fff;
}

nav ul li a {
  text-align: center;
}

nav ul {
  position: relative;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: left;
  list-style-type: none;
  height: 70%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -ms-flex-pack: distribute;
  justify-content: space-around;
}
nav ul li {
  text-align: center;
  margin: 5px 0px;
}
nav a,
nav a:visited,
nav a:active {
  color: #fff;
}
nav a {
  text-decoration: none !important;
  display: inline-block;
  width: 140px;
  padding: 15px 0px;
  position: relative;
  z-index: 0;
  transition: all 0.5s;
}
nav a.active {
  background-color: rgba(0, 0, 0, 0.8);
}
nav a::before {
  content: '';
  position: absolute;
  height: 0%;
  width: 0%;
  bottom: 0;
  left: 0;
  opacity: 1;
  z-index: -1;
}
.logo {
  margin-top: 0px;
  top: 0;
  left: 0;
  z-index: 999999;
  position: fixed;
  display: inline-block;
  text-align: center;
  background-color: #faf5b2;
  opacity: .5;
  height: 15%;
  width: 15%;
}
.logo h1 {
  font-size: 24px;
  text-transform: uppercase;
  font-weight: 900;
  color: #111;
  top: 50%;
  left: 50%;
  position: absolute;
  transform: translate(-50%, -50%);
}
.logo h2 {
  display: none;
}
