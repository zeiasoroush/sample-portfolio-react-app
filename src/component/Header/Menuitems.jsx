import React, { Component } from 'react';
import {  Link } from 'react-router-dom';

class Menuitems extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isactiveId: 1,
      location: 'http://localhost:3000/'
    };
    this.isActive = this.isActive.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }
  isActive(id, index) {
    if (this.state.isactiveId === 0 && index === 0) {
      return 'ui-tabs-active';
    } else {
      return id === this.state.isactiveId ? 'ui-tabs-active' : '';
    }
  }
  handleClick(e, id) {
    const event = e.currentTarget.href;
    this.setState({ isactiveId: id, location: event }, () => {
      this.props.handlecallbackchild(this.state.location);
    });
  }
  render() {
    return (
      <ul>
        {this.props.listcontent.map((content, index) => {
          return (
            <li key={index}>

                <Link
                  className={this.isActive(content.id, index)}
                  onClick={e => this.handleClick(e, content.id)}
                  to={content.link}
                >
                  <img
                    src={require(`../../assets/images/icon-${content.id}.png`)}
                    alt={content.name}
                  />
                  <em>{content.name}</em>
                </Link>

            </li>
          );
        })}
      </ul>
    );
  }
}
export default Menuitems;
