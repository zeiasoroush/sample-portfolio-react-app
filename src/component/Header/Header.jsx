import React, { Component } from 'react';
import './Header.css';
import Menuitems from './Menuitems';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      listItem: [
        {
          id: 1,
          link: '/',
          name: 'HOMEPAGE',
        },
        {
          id: 2,
          link: '/AboutUs',
          name: 'ABOUT US',
        },
        {
          id: 3,
          link: '/Gallery',
          name: 'OUR GALLERY',
        },
        {
          id: 4,
          link: '/ContactUs',
          name: 'CONTACT US',
        }
      ],
      location: ''
    };
    this.handlecallbackchild = this.handlecallbackchild.bind(this);
  }
  handlecallbackchild(location) {
    this.props.callbackhandle(location);
  }

  render() {
    return (
      <div className="Header">
        <div className="logo">
          <h1>Earth</h1>
          <h2>E</h2>
        </div>
        <nav>
          <Menuitems
            listcontent={this.state.listItem}
            handlecallbackchild={this.handlecallbackchild}
          />
        </nav>
      </div>
    );
  }
}

export default Header;
