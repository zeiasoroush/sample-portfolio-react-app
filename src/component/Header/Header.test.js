import React from 'react';
import { shallow } from 'enzyme';
import { stub } from 'sinon';
import Header from './Header';
import Menuitem from './Menuitems';

describe('it should render page header', () => {
  it('should render <Menuitems /> for header items', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
    expect(component.find(Menuitem).length).toEqual(1);
  });
  it('should pass callback from child', () => {
    const handlecallbackchild = jest.fn();
    const wrapper = shallow(
      <Header handlecallbackchild={handlecallbackchild} />
    );
    wrapper.find('Menuitems').prop('handlecallbackchild')(
      'http://localhost:3000/'
    );
    expect(handlecallbackchild.callCount).toEqual(0);
  });
});
