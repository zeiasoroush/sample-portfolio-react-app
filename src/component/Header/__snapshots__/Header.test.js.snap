// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render page header should render <Menuitems /> for header items 1`] = `ShallowWrapper {}`;
